access/validator.go

```diff
@@ -267,13 +267,16 @@ func (v *TransactionValidator) checkAddresses(tx *flow.TransactionBody) error {
 
 // every key (account, key index combination) can only be used once for signing
 func (v *TransactionValidator) checkSignatureDuplications(tx *flow.TransactionBody) error {
-	observedSigs := make(map[string]bool)
+	type uniqueKey struct {
+		address flow.Address
+		index   uint64
+	}
+	observedSigs := make(map[uniqueKey]bool)
 	for _, sig := range append(tx.PayloadSignatures, tx.EnvelopeSignatures...) {
-		keyStr := sig.UniqueKeyString()
-		if observedSigs[keyStr] {
+		if observedSigs[uniqueKey{sig.Address, sig.KeyIndex}] {
 			return DuplicatedSignatureError{Address: sig.Address, KeyIndex: sig.KeyIndex}
 		}
-		observedSigs[keyStr] = true
+		observedSigs[uniqueKey{sig.Address, sig.KeyIndex}] = true
```

Comment:
I assume this change is mostly for performance reasons, could you please check we have tests for this?

Reply:
Yes, it is for performance. Here are the tests https://github.com/onflow/flow-go/blob/master/fvm/transactionVerifier_test.go#L43-L68